### PR TITLE
[Media] Prevent duplicate file names in DB

### PR DESCRIPTION
**Pull request by alisterdev**
- [x] Ensure that no duplicate file names are inserted in the database
- [x] Prompt user when he tries to upload a duplicate file name, advising that he will override existing file.
- [x] Update existing record if user wants to override instead of ignoring DB insert

### CODE REVIEW

File: modules/media/jsx/uploadForm.js [hunk 1]
```diff
@@ -215,12 +216,21 @@ var MediaUploadForm = React.createClass({
...
     if (!this.isValidFileName(requiredFileName, fileName)) {
       alert("File name should begin with: " + requiredFileName);
       return;
     }
 
+    // Check for duplicate file names
+    var isDuplicate = mediaFiles.indexOf(myFormData.file.name);
+    if (isDuplicate >= 0) {
+      var confirmed = confirm(
+        "A file with this name already exists!\n" +
```
Comment (xlecours) [inline modules/media/jsx/uploadForm.js]: This does not work for me.

Comment (xlecours) [inline modules/media/jsx/uploadForm.js]: I was using the unique index on file name in addition to the changes in this pull request.

**Comment by Jkat** [comment]
You should do this for the Data Release module and Document Repository too :) :) :)

**Comment by alisterdev** [comment]
Not in my jurisdiction lol

**Pull request merged**